File: 141.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */
class Solution {
public:
    bool hasCycle(ListNode *head) {
        ListNode *a, *b;
        a=head;
        b=head;
        while (a)
        {
            a=a->next;
            if (a==b)
                return true;
            if (a)
                a=a->next;
            if (a==b)
                return true;
            b=b->next;
        }
        return false;
    }
};

File: 141.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def hasCycle(self, head: ListNode) -> bool:
        a=head
        b=head
        while (a!=None):
            a=a.next;
            if (a==b):
                return True;
            if (a!=None):
                a=a.next;
            if (a==b):
                return True;
            b=b.next;
        return False;
